build_tools/cmake/ctest_all.sh

```diff
@@ -169,7 +171,12 @@ if (( ${#excluded_tests[@]} )); then
   ctest_args+=("--exclude-regex ${excluded_tests_regex}")
 fi
 
-echo "*************** Running CTest ***************"
-
 set -x
+
+echo "*************** Running CTest ***************"
 ctest ${ctest_args[@]}
+
+if (( IREE_CTEST_IREE_DIALECTS_TESTS_DISABLE != 1 )); then
+  echo "**************** Running llvm-external-projects tests ***************"
+  cmake --build ${BUILD_DIR} --target check-iree-dialects -- -k 0
+fi
```

Comment:
Uhhhhh this is rebuilding LLVM over 6 minutes to then run 20 tests in 0.38s. Naaaaaaaaaaah let's not do that.
https://github.com/openxla/iree/actions/runs/7673914344/job/20917770072?pr=16226#step:6:3016

Reply:
Yeah, this won't work with the current setup. As part of https://github.com/openxla/iree/issues/16203 I want to run unit tests as part of the `build_all` job, and that can run these tests (at least until the code is merged with the main tree, https://groups.google.com/g/iree-discuss/c/E2EpE6nFPHw/m/reTX0I-5AwAJ - fyi @hanhanW )